src/main/scala/se/nimsa/sbx/storage/S3Storage.scala

```diff
@@ -33,13 +34,24 @@ import scala.util.control.NonFatal
 
 /**
   * Service that stores DICOM files on AWS S3.
+  *
   * @param s3Prefix prefix for keys
-  * @param bucket S3 bucket
-  * @param region aws region of the bucket
+  * @param bucket   S3 bucket
+  * @param region   aws region of the bucket
   */
-class S3Storage(val bucket: String, val s3Prefix: String, val region: String) extends StorageService {
+class S3Storage(val bucket: String, val s3Prefix: String, val region: String)(implicit system: ActorSystem, materializer: Materializer) extends StorageService {
 
-  val s3Client = new S3Facade(bucket, region)
+  private val settings = new S3Settings(
+    bufferType = MemoryBufferType,
+    diskBufferPath = "",
+    proxy = None,
+    awsCredentials = S3Facade.credentialsFromProviderChain(),
+    s3Region = region,
+    pathStyleAccess = false
+  )
+
+  private val s3Client = new S3Client(settings)
```

Comment:
Better use: def apply(credentials: AWSCredentials, region: String) to create S3Client. When hard coding the settings it is not possible to i.e. use alpakka configuration to enable path style access (which we need) or proxy settings.